corgidrp/spec.py

```diff
@@ -613,8 +613,115 @@ def calibrate_dispersion_model(centroid_psf, band_center_file = None, prism = 'P
         "wavlen_vs_pos_cov": cov_wavlen_vs_pos}
     pri_hdr = centroid_psf.pri_hdr.copy()
     ext_hdr = centroid_psf.ext_hdr.copy()
+    ext_hdr["REFWAVE"] = ref_wavlen
+    ext_hdr["BANDFRAC"] = bandpass_frac
     corgi_dispersion_profile = DispersionModel(
         disp_dict, pri_hdr = pri_hdr, ext_hdr = ext_hdr
     )
 
     return corgi_dispersion_profile
+
+
+def create_wave_cal(disp_model, wave_zeropoint, pixel_pitch_um=13.0):
+    """
+    Create a wavelength calibration map and a wavelength-position lookup table,
+    given a dispersion model and a wavelength zero-point
+
+    Args:
+        disp_model (data.DispersionModel): Dispersion model calibration object
+        wave_zeropoint (dict): Wavelength zero-point dictionary
+        pixel_pitch_um (float): EXCAM pixel pitch in microns
+    
+    Returns:
+        wavlen_map (numpy.ndarray): 2-D wavelength calibration map. Each image
+        pixel value is a wavelength in units of nanometers, computed for the
+        dispersion profile, zero-point position, coordinates, and image shape
+        specified in the input wavelength zero-point object.
+        wavlen_uncertainty (numpy.ndarray): 2-D array of wavelength calibration map
+        uncertainty values in units of nanometers.
+        pos_lookup_table (astropy.table.Table): Wavelength-to-position
+        lookup table, computed for the dispersion profile, zero-point position,
+        coordinates, and image shape specified in the input wavelength
+        zero-point object. The table contains 5 columns: wavelength, x, x
+        uncertainty, y, y uncertainty.
+
+    """
+    ref_wavlen = disp_model.ext_hdr["REFWAVE"]
+    #bandpass_frac = fwhm/cen_wave
+    bandpass_frac = disp_model.ext_hdr["BANDFRAC"]
+    pos_vs_wavlen_poly = np.poly1d(disp_model.pos_vs_wavlen_polycoeff)
+    wavlen_vs_pos_poly = np.poly1d(disp_model.wavlen_vs_pos_polycoeff)
+    wavlen_c = ref_wavlen
+    d_zp_mm = pos_vs_wavlen_poly((wave_zeropoint.get('wavlen') - wavlen_c) / wavlen_c)
+
+    pixel_pitch_mm = pixel_pitch_um * 1E-3
+    theta = np.deg2rad(disp_model.clocking_angle)
+    x_c, y_c = (wave_zeropoint.get('x') - d_zp_mm * np.cos(theta) / pixel_pitch_mm,
+                wave_zeropoint.get('y') - d_zp_mm * np.sin(theta) / pixel_pitch_mm)
+
+    yy, xx = np.indices((wave_zeropoint.get('shapex'), wave_zeropoint.get('shapey')))
+    dd_mm = (xx - x_c) * np.cos(theta) + (yy - y_c) * np.sin(theta) * pixel_pitch_mm
+    wavlen_map = wavlen_vs_pos_poly(dd_mm)
+
+    delta_wav = 0.5
+    n_wav = int(ref_wavlen * bandpass_frac / delta_wav)
+    n_wav_odd = n_wav + (n_wav % 2 == 0) # force odd array length
+    wavlen_beg = ref_wavlen - n_wav_odd // 2 * delta_wav
+    wavlen_end = ref_wavlen + n_wav_odd // 2 * delta_wav
+    
+    wavlens = np.linspace(wavlen_beg, wavlen_end, n_wav_odd)
+    np.testing.assert_almost_equal(wavlens[n_wav_odd // 2], ref_wavlen)
```

Comment:
I generally avoid using asserts in production code (only asserts in tests). Is there a reason we are worried about these not being equal?